packages/components/pie-modal/test/component/pie-modal.spec.ts

```diff
@@ -26,779 +31,763 @@ test.describe('modal', () => {
-            [{
-                mechanism: 'close button',
-                modalCloseFunction: async (page : Page) => {
-                    await page.click(closeButtonSelector);
-                },
-            }, {
-                mechanism: 'Esc key',
-                modalCloseFunction: async (page : Page) => {
-                    await page.keyboard.press('Escape');
-                },
-            }].forEach(({ mechanism, modalCloseFunction }) => {
-                test.describe(`and closed by the ${mechanism}`, () => {
-                    test('should return focus to the element that opens the modal', async ({ page, mount }) => {
-                        // Arrange
-                        const component = renderTestPieModal({ isOpen: false });
-
-                        await mount(WebComponentTestWrapper, {
-                            props: {
-                                pageMode: true,
-                            },
-                            slots: {
-                                component,
-                                pageMarkup: `<div>
-                                    <button id="not-me"></button>
-                                    <button data-test-id="open-modal" id="default"></button>
-                                </div>`,
-                            },
-                        });
+        test.describe('`returnFocusAfterCloseSelector` prop', () => {
+            test.describe('when given', () => {
+                test('should return focus to specified element', async ({ mount, page }) => {
+                    // Arrange
+                    const modalPage = new PieModalPage(page);
 
-                        await page.evaluate(() => {
-                            // Set up a button which opens the modal when clicked
-                            document.querySelector('[data-test-id="open-modal"]')?.addEventListener('click', () => {
-                                document.querySelector('pie-modal')?.setAttribute('isOpen', 'true');
+                    const component = renderTestPieModal({
+                        returnFocusAfterCloseSelector: '#focus-me',
+                    });
+
+                    await mount(WebComponentTestWrapper, {
+                        props: {
+                            pageMode: true,
+                        },
+                        slots: {
+                            component,
+                            pageMarkup: `<div>
+                                <button id="default"></button>
+                                <button id="focus-me"></button>
+                                <button id="not-me"></button>
+                            </div>`,
+                        },
+                    });
+
+                    // Act
+                    await modalPage.clickCloseModal();
+
+                    const focusedElement = await page.locator(':focus');
+                    const focusedElementId = await focusedElement.getAttribute('id');
```

Comment:
I know this code existed before the change, but should probably see if we can get away from doing Locator specific stuff like this in the test specs.